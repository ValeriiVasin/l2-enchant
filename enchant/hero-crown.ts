import { Enchant, EnchantMap, Item } from '../types';

const chances = [60, 50, 40, 35, 30, 28, 26, 24, 22, 20];
const items = [
  Item.HeroCrown,
  Item.HeroCrown_1,
  Item.HeroCrown_2,
  Item.HeroCrown_3,
  Item.HeroCrown_4,
  Item.HeroCrown_5,
  Item.HeroCrown_6,
  Item.HeroCrown_7,
  Item.HeroCrown_8,
  Item.HeroCrown_9,
  Item.HeroCrown_10,
];

export const heroCrownEnchantMap: EnchantMap = new Map(
  items.slice(0, -1).map((item, index): [Item, Enchant] => {
    return [
      item,
      {
        item,
        required: Item.Nothing,
        success: items[index + 1],
        fail: Item.Nothing,
        successRate: chances[index],
      },
    ];
  })
);
