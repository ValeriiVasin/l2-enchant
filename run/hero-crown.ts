import { run } from '../helpers/run';
import { Item } from '../items';

run(
  Item.HeroCrown,
  [
    Item.HeroCrown_2,
    Item.HeroCrown_3,
    Item.HeroCrown_4,
    Item.HeroCrown_5,
    Item.HeroCrown_6,
    Item.HeroCrown_7,
    Item.HeroCrown_8,
    Item.HeroCrown_9,
    Item.HeroCrown_10,
  ],
  100_000,
);
