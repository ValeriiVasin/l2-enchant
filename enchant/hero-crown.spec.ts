import { heroCrownEnchantMap } from './hero-crown';

it('enchant map: hero crown', () => {
  expect(heroCrownEnchantMap).toMatchSnapshot();
});
