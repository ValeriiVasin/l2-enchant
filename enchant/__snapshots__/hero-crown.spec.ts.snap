// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enchant map: hero crown 1`] = `
Map {
  "Венец Героя" => Object {
    "cost": "Nothing",
    "item": "Венец Героя",
    "results": Array [
      Object {
        "chance": 60,
        "item": "Венец Героя +1",
      },
      Object {
        "chance": 40,
        "item": "Nothing",
      },
    ],
  },
  "Венец Героя +1" => Object {
    "cost": "Nothing",
    "item": "Венец Героя +1",
    "results": Array [
      Object {
        "chance": 50,
        "item": "Венец Героя +2",
      },
      Object {
        "chance": 50,
        "item": "Nothing",
      },
    ],
  },
  "Венец Героя +2" => Object {
    "cost": "Nothing",
    "item": "Венец Героя +2",
    "results": Array [
      Object {
        "chance": 40,
        "item": "Венец Героя +3",
      },
      Object {
        "chance": 60,
        "item": "Nothing",
      },
    ],
  },
  "Венец Героя +3" => Object {
    "cost": "Nothing",
    "item": "Венец Героя +3",
    "results": Array [
      Object {
        "chance": 35,
        "item": "Венец Героя +4",
      },
      Object {
        "chance": 65,
        "item": "Nothing",
      },
    ],
  },
  "Венец Героя +4" => Object {
    "cost": "Nothing",
    "item": "Венец Героя +4",
    "results": Array [
      Object {
        "chance": 30,
        "item": "Венец Героя +5",
      },
      Object {
        "chance": 70,
        "item": "Nothing",
      },
    ],
  },
  "Венец Героя +5" => Object {
    "cost": "Nothing",
    "item": "Венец Героя +5",
    "results": Array [
      Object {
        "chance": 28,
        "item": "Венец Героя +6",
      },
      Object {
        "chance": 72,
        "item": "Nothing",
      },
    ],
  },
  "Венец Героя +6" => Object {
    "cost": "Nothing",
    "item": "Венец Героя +6",
    "results": Array [
      Object {
        "chance": 26,
        "item": "Венец Героя +7",
      },
      Object {
        "chance": 74,
        "item": "Nothing",
      },
    ],
  },
  "Венец Героя +7" => Object {
    "cost": "Nothing",
    "item": "Венец Героя +7",
    "results": Array [
      Object {
        "chance": 24,
        "item": "Венец Героя +8",
      },
      Object {
        "chance": 76,
        "item": "Nothing",
      },
    ],
  },
  "Венец Героя +8" => Object {
    "cost": "Nothing",
    "item": "Венец Героя +8",
    "results": Array [
      Object {
        "chance": 22,
        "item": "Венец Героя +9",
      },
      Object {
        "chance": 78,
        "item": "Nothing",
      },
    ],
  },
  "Венец Героя +9" => Object {
    "cost": "Nothing",
    "item": "Венец Героя +9",
    "results": Array [
      Object {
        "chance": 20,
        "item": "Венец Героя +10",
      },
      Object {
        "chance": 80,
        "item": "Nothing",
      },
    ],
  },
}
`;
